### PR TITLE
feat: add agent for collect machine metrics

### LINKED ISSUE #109: [bug] CPU info for Windows OS doesn't work
See:

N15{ 6LFKEF %__6~ZW07 D

### PR DESCRIPTION
**Pull request by leo220yuyaodog**
Fix: https://github.com/casvisor/casvisor/issues/109

### CODE REVIEW

File: controllers/base.go [hunk 1]
```diff
@@ -26,6 +26,10 @@ type ApiController struct {
 	beego.Controller
 }
 
+type RootController struct {
```
Comment (hsluoyz) [inline controllers/base.go:29]: Why don't use ApiController?

File: controllers/agent.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+func (c *RootController) GetSystemInfo() {
+	systemInfo, err := object.GetSystemInfo()
+	if err != nil {
+		c.ResponseError(err.Error())
+		return
+	}
+
+	c.Data["json"] = systemInfo
+	c.ServeJSON()
```
Comment (hsluoyz) [inline controllers/agent.go]: Why don't use c.ResponseOk()?

File: routers/router.go [hunk 3]
```diff
@@ -78,4 +85,6 @@ func initAPI() {
 	beego.Router("/api/update-permission", &controllers.ApiController{}, "POST:UpdatePermission")
 	beego.Router("/api/add-permission", &controllers.ApiController{}, "POST:AddPermission")
 	beego.Router("/api/delete-permission", &controllers.ApiController{}, "POST:DeletePermission")
+
+	beego.Router("/agent/get-system-info", &controllers.RootController{}, "GET:GetSystemInfo")
```
Comment (hsluoyz) [inline routers/router.go:89]: Why not use /api/ prefix?

File: main.go [hunk 4]
```diff
@@ -42,10 +42,9 @@ func main() {
 
 	beego.SetStaticPath("/swagger", "swagger")
 	// https://studygolang.com/articles/2303
-	beego.InsertFilter("/", beego.BeforeRouter, routers.TransparentStatic) // must has this for default page
-	beego.InsertFilter("/*", beego.BeforeRouter, routers.TransparentStatic)
+	beego.InsertFilter("*", beego.BeforeRouter, routers.TransparentStatic)
 	beego.InsertFilter("/dbgate/*", beego.BeforeRouter, routers.ProxyFilter)
-	beego.InsertFilter("/api/*", beego.BeforeRouter, routers.ApiFilter)
+	beego.InsertFilter("*", beego.BeforeRouter, routers.ApiFilter)
```
Comment (hsluoyz) [inline main.go:47]: Why?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: controllers/base.go (hunk 1 above)
Comment (leo220yuyaodog) [reply controllers/base.go:29]: To distinguish it from the ApiController, the api prefix is used between the front and back ends

File: controllers/agent.go (hunk 2 above)
Comment (leo220yuyaodog) [reply controllers/agent.go]: Easy to deserialize the systeminfo struct

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**